src/mixins/raster-layer-line-mixin.js

```diff
@@ -0,0 +1,869 @@
+            rndrProp
+          )
+        }
+      })
+    }
+  }
+
+  _layer._areResultsValidForPopup = function(results) {
+    if (state.encoding.geocol && results[state.encoding.geocol]) {
+      return true
+    }
+    return false
+  }
+
+
+  class PolySvgFormatter {
+    /**
+     * Builds the bounds from the incoming poly data
+     * @param {AABox2d} out AABox2d to return
+     * @param {object} data Object with return data from getResultRowForPixel()
+     * @param {Number} width Width of the visualization div
+     * @param {Number} height Height of the visualization div
+     * @param {object} margin Margins of the visualization div
+     * @param {Function} xscale d3 scale in x dimension from world space to pixel space (i.e. mercatorx-to-pixel)
+     * @param {Function} yscale d3 scale in y dimension from world space to pixel space (i.e. mercatory-to-pixel)
+     */
+    getBounds(data, width, height, margins, xscale, yscale) {
+      throw new Error("This must be overridden")
+    }
+
+    /**
+     * Builds the svg path string to use with the d svg attr:
+     * https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d
+     * This function should be called after the getBounds().
+     * The t/s arguments are the transformations to properly place the points underneath
+     * a parent SVG group node. That node is what ultimately handles animations and such
+     * so we need to transform all the points into local space. t is the translation
+     * and s is the scale to transform the points from pixel space to model/object space.
+     * @param {string} out Returns the svg path string
+     * @param {Point2d} t Translation from world to object space.
+     * @param {Number} s Scale from world to object space.
+     */
+    getSvgPath(t, s) {
+      throw new Error("This must be overridden")
+    }
+  }
+
+  function buildGeoProjection(
+    width,
+    height,
+    margins,
+    xscale,
+    yscale,
+    clamp = true,
+    t = [0, 0],
+    s = 1
+  ) {
+    let _translation = t,
+      _scale = s,
+      _clamp = clamp
+
+    const project = d3.geo.transform({
+      point(lon, lat) {
+        const projectedCoord = conv4326To900913(lon, lat)
+        const pt = [
+          _scale * (xscale(projectedCoord[0]) + margins.left - _translation[0]),
+          _scale *
+          (height -
+            yscale(projectedCoord[1]) -
+            1 +
+            margins.top -
+            _translation[1])
+        ]
+        if (_clamp) {
+          if (pt[0] >= 0 && pt[0] < width && pt[1] >= 0 && pt[1] < height) {
+            return this.stream.point(pt[0], pt[1])
+          }
+        } else {
+          return this.stream.point(pt[0], pt[1])
+        }
+      }
+    })
+
+    project.setTransforms = (t, s) => {
+      _translation = t
+      _scale = s
+    }
+
+    project.setClamp = clamp => {
+      _clamp = Boolean(clamp)
+    }
+
+    return project
+  }
+
+  class GeoSvgFormatter extends PolySvgFormatter {
```

Comment:
Looks like this is exactly the same as in the poly layer. Can we move this into a utility module somewhere to reduce duplicate code?